internal/zero/healthcheck/syncer.go

```diff
@@ -0,0 +1,51 @@
+package healthcheck
+
+import (
+	"context"
+
+	configpb "github.com/pomerium/pomerium/pkg/grpc/config"
+	"github.com/pomerium/pomerium/pkg/grpc/databroker"
+	"github.com/pomerium/pomerium/pkg/health"
+	"github.com/pomerium/pomerium/pkg/protoutil"
+)
+
+func (c *checker) ConfigSyncer(ctx context.Context) error {
```

Comment:
Why should the health checks sync config separately from the rest of the app?

What are the difference between this approach compared with registering an OnConfigChange() listener on the main config source?

Reply:
that's a fair point, however here we only have access to the lowest layer of config. 
[by design](https://github.com/pomerium/pomerium/blob/7d8af09036fe0e6de0122c3c490393d1da68f4c5/config/config_source.go#L49-L53) higher layers are notified when lower layer configs are changed but not the other way around. 

we would have to move this health check to some entirely different place to be able to attach to a regular config.Source; there are benefits doing that; I can make a separate PR for that.